JS OOP/JS OOP Methods, Properties/index.js

```diff
@@ -0,0 +1,217 @@
+        return first;
+    }
+
+    getLast() {
+        let last = this.head;
+        if(last) {
+            while(last.next) {
+                last = last.next;
+            }
+        }
+        return last.el;
+    }
+
+    getLength() {
+        return this.length;
+    }
+
+    append = (...el) => {
+        el.forEach(el => {
+            const node = new listNode(el);
+
+            if(this.head === null) {
+                this.head = node
+            } else {
+                let previous = this.head;
+                while(previous.next) {
+                    previous = previous.next;
+                }
+                previous.next = node;
+            }
+            this.length++;
+        })
+        return this;
+    }
+
+    prepend = (...el) => {
+        el.reverse().forEach(el => {
+            const node = new listNode(el);
+
+            if(this.head === null) {
+                this.head = node;
+            } else {
+                node.next = this.head;
+                this.head = node;
+            }
+            this.length++;
+        })
+        
+        return this;
+    }
+
+    insert = (index, ...el) => {
+        el.reverse().forEach(el => {
+            if(index < 0 || index > this.length) {
+                console.log("Invalid index");
+                return;
+            } else if(index === 0) {
+                this.prepend(el);
+            } else {
+                const node = new listNode(el);
+                let previous = this.head;
+                for(let i = 0; i < index - 1; i++) {
+                    previous = previous.next;
+                }
+                node.next = previous.next;
+                previous.next = node;
+                this.length++;
+            }
+        })
+        return this;
+    }
+
+    removeAt(index) {
+        let removedNode;
+        if(index < 0 || index > this.length) {
+            return "Invalid index";
+        } else if (index === 0) {
+            removedNode = this.head;
+            this.head = this.head.next;
+            this.length--;
+        } else {
+            let previous = this.head;
+            for(let i = 0; i < index - 1; i++) {
+                previous = previous.next;
+            }
+            removedNode = previous.next;
+            previous.next = removedNode.next;
+            this.length--;
+            return `The element at index ${index} -> (${removedNode.el}) has been successfully removed`;
+        }
+    }
+
+    at(index, el) {
+        if(index < 0 || index > this.length) {
+            console.log("Invalid index");
+            return;
+        } else {
+            let previous = this.head;
+            for(let i = 0; i < index; i++) {
+                previous = previous.next;
+            } 
+            if(el != undefined) {
+                previous.el = el;
+                return el;
+            } 
+            else {
+                return `The element at index ${index} is ${previous.el}` 
```

Comment:
at function should return the element at the passed index or throw an error if the index is incorrect.